source/utils.py

```diff
@@ -0,0 +1,14 @@
+from basicauth import decode
+from models import *
+from werkzeug.exceptions import BadRequest, NotFound
+
+def authorization(encoded_str):
+    if encoded_str == None:
+        raise BadRequest('Authorization failed')
+    username, password = decode(encoded_str)
+    user = User.query.filter_by(username=username).first()
+    if user == None:
+        raise NotFound('User not found')
+    elif user.password == password:
```

Comment:
It's best to validate if they are not the same, raise the exception within the `elif` block and return user as the last instance of the method